aggregators/aggregator.go

```diff
@@ -603,23 +603,134 @@ func (a *Aggregator) processHarvest(
 	hs := harvestStats{
 		eventsTotal:            cm.EventsTotal,
 		youngestEventTimestamp: modelpb.ToTime(cm.YoungestEventTimestamp),
-		servicesOverflowed:     hllSketchEstimate(cm.OverflowServicesEstimator),
 	}
-	addOverflow := func(o *aggregationpb.Overflow) {
+
+	a.getAndLogOverflowStats(&hs, cm, aggIvl)
+
+	if err := a.cfg.Processor(ctx, cmk, cm, aggIvl); err != nil {
+		return harvestStats{}, fmt.Errorf("failed to process combined metrics ID %s: %w", cmk.ID, err)
+	}
+	return hs, nil
+}
+
+func (a *Aggregator) getAndLogOverflowStats(hs *harvestStats, cm *aggregationpb.CombinedMetrics, aggIvl time.Duration) {
+	hs.servicesOverflowed = hllSketchEstimate(cm.OverflowServicesEstimator)
+
+	logFunc := func(msg string, fields ...zap.Field) {
+		if a.cfg.OverflowLogging.Func != nil &&
+			(a.cfg.OverflowLogging.AggregationInterval == 0 || a.cfg.OverflowLogging.AggregationInterval == aggIvl) {
+			a.cfg.OverflowLogging.Func(msg, fields...)
+		}
+	}
+
+	// TODO(carsonip): Update this log message when global labels implementation changes
+	logFunc(fmt.Sprintf(""+
+		"Service limit reached, new metric documents will be grouped under a dedicated "+
```

Comment:
Maybe these would be a bit more readable as package-level vars, using raw (back quote) strings?

Reply:
Back quotes are no good because we don't want newlines, but I'll still move them to package-level vars